contrib/babelfishpg_unit/README.md

```diff
@@ -0,0 +1,66 @@
+# How to add unit tests for babelfish?
+
+Babelfish has introduced a new extension named babelfishpg_unit which enables us to run unit tests. Please follow the README.md file present in contrib/README.md. Once we have built and installed the extension, we are ready to write unit tests. 
+
+* To maintain a well-organized structure, the directory layout will be as follows:
+
+- babelfish_extensions
+    -- contrib
+        --- babelfishpg_common
+        --- babelfishpg_money
+        --- babelfishpg_tsql
+        --- babelfishpg_tds
+        --- babelfishpg_unit
+            ---- Makefile
+            ---- babelfishpg_unit.control
+            ---- babelfishpg_unit--1.0.0.sql 
+            ---- babelfishpg_unit.c 
+            ---- babelfishpg_unit.h 
+            ---- test_1.c
+            ---- test_2.c
+
+As we can see, test_1.c, test_2.c, ... are the individual test files which contains the test cases to a particular area of functionality or component being tested.
+
+
+- To add a new test to the existing framework, within the framework's directory, create a new .c file and name it according to the test we want to add (e.g., test_1.c). This file will contain the code for our new test. By organising the tests into separate files, it helps maintain a modular and structured approach. Each file can focus on a specific test or group of related tests, making it easier to manage and maintain the overall framework.
+    - Declare the function you want to test as extern in babelfishpg_unit.h. Eg:
+        - extern TestResult* test_int4_fixeddecimal_ge(void)
+    - Metadata for each test function is added as a row in the tests array, containing the necessary information for the test. This tests array is located in babelfishpg_unit.c. Eg:
+        - TestInfo tests[]=
+            {
+                {&test_int4_fixeddecimal_ge, true, "GreaterThanOrEqualToCheck_INT4_FIXEDDECIMAL", "babelfish_money_datatype"},
+            };
+    - It is important to note that the TestInfo struct is a data structure used to store metadata information about a test function within a testing framework. So, the return type of all functions which are to be tested must be TestInfo*.
+    - Every function should have some expected and obtained output. One should use TEST_ASSERT_TESTCASE(expected == obtained, testResult) first and then TEST_ASSERT(expected == obtained, testResult) to obtain the status of a test. In the TEST_ASSERT_TESTCASE marco, we only set the result field of TestResult struct to true/false based on result. In TEST_ASSERT, we decide whether all the tests have passed or not. Eg:
+        - TestResult* test_int4_fixeddecimal_ge(void)
```

Comment:
We should render this as a code snippet?

Reply:
Done!